d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperAnnouncer.java

```diff
@@ -629,4 +700,65 @@ public boolean isMarkUpFailed()
     return _markUpFailed;
   }
 
+  @Override
+  public void emitSDStatusInitialRequestEvent(String cluster, long duration, boolean succeeded) {
+    if (_eventEmitter == null) {
+      _log.warn("Service discovery event emitter in ZookeeperAnnouncer is null. Skipping emitting events.");
```

Comment:
I'm thinking of adding the emitter with the container part code. But it could be a problem if the app bump up only pegasus without the container part. I'll change it and add in the message to reminder bumping up container.

Reply:
we probably shouldn't mention "container" in pegasus comment? :(